percona/controllers/pgc/pgc.go

```diff
@@ -285,3 +291,52 @@ func (c *Controller) AddPerconaPGClusterEventHandler() {
 func (c *Controller) WorkerCount() int {
 	return c.PerconaPGClusterWorkerCount
 }
+
+func PrepareForRestore(clientset *kubeapi.Client, clusterName, namespace string) error {
+	ctx := context.TODO()
+	err := deleteDatabasePods(clientset, clusterName, namespace)
+	if err != nil {
+		log.Error("get pods: %s", err.Error())
+	}
+	// Delete the DCS and leader ConfigMaps.  These will be recreated during the restore.
+	configMaps := []string{
+		fmt.Sprintf("%s-config", clusterName),
+		fmt.Sprintf("%s-leader", clusterName),
+	}
+	for _, c := range configMaps {
+		if err := clientset.CoreV1().ConfigMaps(namespace).
+			Delete(ctx, c, metav1.DeleteOptions{}); err != nil && !kerrors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func deleteDatabasePods(clientset *kubeapi.Client, clusterName, namespace string) error {
+	ctx := context.TODO()
+	pgInstances, err := clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s,%s", config.LABEL_PG_CLUSTER, clusterName,
+			config.LABEL_PG_DATABASE),
+	})
+	if err != nil {
```

Comment:
we need to check if err there is not `NotFound` error.
Because in case of no pods, we will return error